encodings/dict/src/arrow.rs

```diff
@@ -17,3 +21,40 @@ impl<K: ArrowDictionaryKeyType> FromArrowArray<&DictionaryArray<K>> for DictArra
         DictArray::try_new(keys, values).vortex_unwrap()
     }
 }
+
+inventory::submit!(ToArrowKernelAdapter(DictVTable).lift());
```

Comment:
We have a wrapper macro `register_kernel!` in the vortex_array crate that I think you'll want instead

https://docs.rs/vortex-array/latest/vortex_array/macro.register_kernel.html